pkg/portallocator/portallocator.go

```diff
@@ -56,18 +51,34 @@ type PortAllocator struct {
 	nodeInformer       cache.SharedIndexInformer
 }
 
-// NewPortAllocator returns a new dynamic port
+// portallocator.Interface manages the dynamic port allocation strategy.
+//
+// The portallocator does not currently support mixing static portAllocations (or any pods with defined HostPort)
+// within the dynamic port range other than the ones it coordinates.
+type Interface interface {
```

Comment:
nit: Since this is the exported type for this package, can you move it above the non-exported concrete type?

Reply:
Done